academy/xl-cosmjs/first-steps.md

```diff
@@ -0,0 +1,637 @@
+
+When you finally launch `simd`, you see the line:
+
+```sh
+$ ./build/simd start
+...
+4:37PM INF Starting RPC HTTP server on 127.0.0.1:26657 module=rpc-server
+...
+```
+
+This is the port to which you connect CosmJs.
+
+## Prepare for `simd`
+
+You can reuse your `experiment.ts` script, with some adjustments. Although you have Alice's address, you may not have her mnemonic or private key. The private key is stored in your operating system's keyring backend. For the purpose of this exercise you can extract it, but generally this is an unsafe operation:
+
+```sh
+$ ./build/simd keys export alice --unsafe --unarmored-hex
+```
+
+You get a 64-digit-long hex value. Copy-paste it into a new `simd.alice.private.key` file in your `cosmjs-template` folder. The `.gitignore` was already configured earlier to ignore it, which mitigates the risk.
+
+<HighlightBox type="tip">
+
+If you cannot remember which alias you gave your keys, `list` them:
+
+```sh
+$ ./build/simd keys list
+- address: cosmos1c3srguwnzah5nd4cn49shltvr6tsrcl2jwn8je
+  name: alice
+  pubkey: '{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"AhR7SWWDsaSxBD9r/mIhbVOWap70jA3WpBIqjOJo4Dwp"}'
+  type: local
+- address: cosmos1umpxwaezmad426nt7dx3xzv5u0u7wjc0kj7ple
+  name: bob
+  pubkey: '{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"Av1VW23/laXWtbwWwOUHCvjjeLLqbdzazRneeRsE/shL"}'
+  type: local
+```
+
+</HighlightBox>
+
+## Update your script
+
+With the new elements in place, you can update your script. Change `rpc` and set `faucet` to Bob's address:
+
+```typescript
+const rpc = "http://127.0.0.1:26657"
+const faucet = "cosmos1umpxwaezmad426nt7dx3xzv5u0u7wjc0kj7ple"
+```
+
+Next, add a function to create Alice's signer. In VSCode, right-click on `OfflineDirectSigner` again and select <kbd>Find All Implementations</kbd>: `DirectSecp256k1Wallet` is the more appropriate choice this time.
+
+Add the import:
+
+```typescript
+import { DirectSecp256k1Wallet, OfflineDirectSigner } from "@cosmjs/proto-signing"
+```
+
+In `DirectSecp256k1Wallet` the `fromKey` factory function needs a `Uint8Array`. Fortunately, CosmJs includes a utility to convert a hexadecimal string into a `Uint8Array`. Import it:
+
+```typescript
+import { fromHex } from "@cosmjs/encoding"
+```
+
+Now create a new function to get a signer:
+
+```typescript
+const getAliceSignerFromPriKey = async(): Promise<OfflineDirectSigner> => {
+    return DirectSecp256k1Wallet.fromKey(
+        fromHex((await readFile("./simd.alice.private.key")).toString()),
+        "cosmos",
+    )
+}
+```
+
+The address prefix is `"cosmos"`, as can be seen in Alice's address. Insert it to replace `getAliceSignerFromMnemonic`. Also change the token unit from `uatom` to `stake`. Experiment with adjusting the values as desired. Run it and confirm the output is as expected.
```

Comment:
```suggestion
The address prefix is `"cosmos"`, as can be seen in Alice's address. Insert it to replace `getAliceSignerFromMnemonic`. Also change the token unit from `uatom` to `stake`, because this is the default token when using `simapp`. Experiment with adjusting the values as desired. Run it and confirm the output is as expected.
```